### PR TITLE
Override different serve definition in nanohttpd

### PR DESCRIPTION
**Pull request by dpgraham**
* There appeared to be a race condition that was causing multiple routes to have the same POST body (Isaac had this problem during the hack week)
* This way appears to fix it and is cleaner too

### CODE REVIEW

File: espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/http/Server.java [hunk 1]
```diff
@@ -21,7 +22,7 @@
     public Server() throws IOException, DuplicateRouteException {
         super(8080);
```
Comment (mykola-mokhnach) [inline espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/http/Server.java]: can we move port number to a constant?

File: espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/http/Server.java [hunk 2]
```diff
@@ -34,11 +35,11 @@ public Server() throws IOException, DuplicateRouteException {
     }
 
     @Override
-    public Response serve(IHTTPSession session) {
+    public Response serve(String uri, Method method, Map<String, String> headers, Map<String, String> parms, Map<String, String> files) {
```
Comment (mykola-mokhnach) [inline espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/http/Server.java]: typo -> params

File: espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/http/Router.java [hunk 3]
```diff
@@ -143,13 +140,7 @@ public BaseResponse route(IHTTPSession session) {
 
         // Parse the appium params
         String postJson;
```
Comment (mykola-mokhnach) [inline espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/http/Router.java]: you can combine variable definition and initialization now

File: espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/http/Server.java (hunk 2 above)
Comment (dpgraham) [inline espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/http/Server.java]: The typo is in the NanoHTTPD. This is overriding a method.

**Comment by dpgraham** [comment]
Fixes made

File: espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/http/Server.java (hunk 2 above)
Comment (mykola-mokhnach) [inline espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/http/Server.java]: I assume in strictly-typed Java world only argument types matter, but you can use your own names for override method arguments. 
https://stackoverflow.com/questions/9627139/overriding-methods-with-different-variable-names

**Pull request merged**